gateway/ln-gateway/src/lib.rs

```diff
@@ -384,28 +384,27 @@ impl Gateway {
             );
         }
 
-        let federation_info = actor.get_info()?;
+        let federation_info = actor.get_summary_info()?;
 
         Ok(federation_info)
     }
 
     async fn handle_get_info(&self, payload: InfoPayload) -> Result<GatewayInfo> {
-        let actors = self.actors.read().await;
-        let mut all_federations: Vec<FederationInfo> = Vec::new();
-        for actor in actors.values() {
-            all_federations.push(actor.get_info()?);
-        }
-
-        let federations = payload
-            .federation_id
-            .map(|id| {
-                all_federations
-                    .iter()
-                    .filter(|info| info.federation_id.eq(&id))
-                    .cloned()
-                    .collect::<Vec<_>>()
-            })
-            .unwrap_or(all_federations);
+        let mut federations: Vec<FederationInfo> = Vec::new();
+        match payload.federation_id {
+            Some(federation_id) => federations.push(
+                self.select_actor(federation_id)
+                    .await?
+                    .get_detailed_info()
+                    .await?,
+            ),
+            None => {
+                let actors = self.actors.read().await;
+                for actor in actors.values() {
+                    federations.push(actor.get_summary_info()?);
+                }
+            }
+        };
```

Comment:
Never mind my previous comment, fixed here. When it comes to PR structure though: it would be nice not to undo changes of previous commits, otherwise a lot of time can be wasted when reviewing commit-by-commit. Instead prefer interactive rebasing/amending commits when you notice something has to be changed in the history.

Reply:
Sorry about that 😅, I should've squashed/amended some changes. Thanks for the tip, I'll keep that in mind for the next PRs